pageserver/src/remote_storage/storage_sync/download.rs

```diff
@@ -69,31 +69,36 @@ pub(super) async fn download_timeline<
             return DownloadedTimeline::Abort;
         }
         Some(index_entry) => match index_entry {
-            TimelineIndexEntry::Full(remote_timeline) => Cow::Borrowed(remote_timeline),
+            TimelineIndexEntry::Full(full_entry) => Cow::Borrowed(&full_entry.remote_timeline),
             TimelineIndexEntry::Description(_) => {
                 let remote_disk_consistent_lsn = index_entry.disk_consistent_lsn();
+                let timeline_awaits_download = index_entry.get_awaits_download();
```

Comment:
Is it so, that we don't actually prevent concurrent downloads of the same timeline with this field?
Feels like we should abort here on `true`, shouldn't we? 

In a way, we're protected by the `attach_timeline` handle logic if it's implemented right, but such data races are IMO bad enough to protect from them here too.

Reply:
Added a check here

Reply:
Actually check here is wrong, because this code can be is triggered for the same first download. In attach call we set awaits download to true and then when this code is executed awaits download here being true doesnt stand for a concurrent download